presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -122,6 +124,22 @@ default ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession s
         throw new PrestoException(GENERIC_INTERNAL_ERROR, "ConnectorMetadata getCommonPartitioningHandle() is implemented without getAlternativeLayout()");
     }
 
+    /**
+     * Experimental: if true, the engine will invoke pushdownFilter instead of getTableLayouts.
+     */
+    default boolean isPushdownFilterSupported(ConnectorSession session, ConnectorTableHandle tableHandle)
```

Comment:
Maybe add a comment saying that it will be replaced once `FilterNode` is in SPI (#12546)

Reply:
Curious: Why this SPI needs to have `tableHandle`? 

- For now it returns true for partitioned table, and false for unpartitioned table. However the predicate being pushed down might not include partition columns. So this check doesn't look to me to be necessary? 
- In the future we can pushdown complex type extractions, so any table with complex type is candidate for pushdown filter.

Reply:
@wenleix In this iteration of Aria, filter pushdown is supported only for tables stored using ORC or DWRF formats. In the future, we hope to add support for Parquet. Technically speaking, I could move that check into the pushdownFilter method itself, but it seems cleaner and safer to have it here.